xrspatial/tests/general_checks.py

```diff
@@ -29,23 +48,65 @@ def general_output_checks(input_agg: xr.DataArray,
             )
 
     if expected_results is not None:
-        numpy_func = lambda output, expected: np.testing.assert_allclose(  # noqa: E731, E501
-            output, expected_results, equal_nan=True, rtol=1e-06
-        )
-        dask_func = lambda output, expected: np.testing.assert_allclose(  # noqa: E731, E501
-            output.compute(), expected_results, equal_nan=True, rtol=1e-06
-        )
-        cupy_func = lambda output, expected: np.testing.assert_allclose(  # noqa: E731, E501
-            output.get(), expected_results, equal_nan=True, rtol=1e-06
-        )
-        dask_cupy_func = lambda output, expected: np.testing.assert_allclose(  # noqa: E731, E501
-            output.compute().get(), expected_results,
-            equal_nan=True, rtol=1e-06
-        )
+        get_numpy_data = lambda output: output  # noqa: E731
+        get_dask_numpy_data = lambda output: output.compute()  # noqa: E731
+        get_cupy_data = lambda output: output.get()  # noqa: E731
+        get_dask_cupy_data = lambda output: output.compute().get()  # noqa: E731
+
         mapper = ArrayTypeFunctionMapping(
-            numpy_func=numpy_func,
-            dask_func=dask_func,
-            cupy_func=cupy_func,
-            dask_cupy_func=dask_cupy_func,
+            numpy_func=get_numpy_data,
+            dask_func=get_dask_numpy_data,
+            cupy_func=get_cupy_data,
+            dask_cupy_func=get_dask_cupy_data,
         )
-        mapper(output_agg)(output_agg.data, expected_results)
+        output_data = mapper(output_agg)(output_agg.data)
+        np.testing.assert_allclose(output_data, expected_results, equal_nan=True, rtol=rtol)
+
+        if verify_dtype:
+            assert output_data.dtype == expected_results.dtype
+
+
+def assert_nan_edges_effect(result_agg):
+    # nan edge effect
+    edges = [
+        result_agg.data[0, :],
+        result_agg.data[-1, :],
+        result_agg.data[:, 0],
+        result_agg.data[:, -1],
+    ]
+    for edge in edges:
+        np.testing.assert_allclose(
+            edge, np.full(edge.shape, np.nan), equal_nan=True

```

Comment:
Because we are asserting that all values are equal to `np.nan` we can use the simpler
```python
np.testing.assert_array_equal(edge, np.nan)
```
Unless I have missed some subtlety when using `cupy` or `dask`.